lib/api/core/validation/index.js

```diff
@@ -1,690 +1,708 @@
     }
-    else if (collectionSpecSubset[fieldName].defaultValue && isUpdate && documentSubset.hasOwnProperty(fieldName) && documentSubset[fieldName] === null) {
-      documentSubset[fieldName] = collectionSpecSubset[fieldName].defaultValue;
-    }
-  });
-
-  return documentSubset;
-};
 
-/**
- * @param {*} documentSubset
- * @param {StructuredFieldSpecification} collectionSpecSubset
- * @param {boolean} strictness
- * @param {string[]} errorMessages
- * @param {boolean} verbose
- */
-Validation.prototype.recurseFieldValidation = function validationRecurseFieldValidation (documentSubset, collectionSpecSubset, strictness, errorMessages, verbose) {
-  if (strictness) {
-    if (!checkAllowedProperties(documentSubset, Object.keys(collectionSpecSubset))) {
-      // We use a throw to be able to provide information about the field or the document in whole
-      throw new BadRequestError('strictness');
+    if (!verbose) {
+      // We stop as soon as one field is not valid
+      return Object.keys(collectionSpecSubset).every(fieldName =>
+        this.isValidField(fieldName, documentSubset, collectionSpecSubset, strictness, errorMessages, verbose)
+      );
     }
-  }
 
-  if (!verbose) {
-    // We stop as soon as one field is not valid
-    return Object.keys(collectionSpecSubset).every(fieldName => {
-      return this.isValidField(fieldName, documentSubset, collectionSpecSubset, strictness, errorMessages, verbose);
-    });
+    // We try to validate every field in order to get all error messages if any
+    return Object.keys(collectionSpecSubset).reduce((reductionResult, fieldName) => {
+      return this.isValidField(fieldName, documentSubset, collectionSpecSubset, strictness, errorMessages, verbose) && reductionResult;
+    }, true);
   }
 
-  // We try to validate every field in order to get all error messages if any
-  return Object.keys(collectionSpecSubset).reduce((reductionResult, fieldName) => {
-    return this.isValidField(fieldName, documentSubset, collectionSpecSubset, strictness, errorMessages, verbose) && reductionResult;
-  }, true);
-};
+  /**
+   * @param {string} fieldName
+   * @param {*} documentSubset
+   * @param {StructuredFieldSpecification} collectionSpecSubset
+   * @param {boolean} strictness
+   * @param {string[]} errorMessages
+   * @param {boolean} verbose
+   * @returns {boolean}
+   */
+  isValidField(fieldName, documentSubset, collectionSpecSubset, strictness, errorMessages, verbose) {
+    var
+      nestedStrictness,
+      /** @type StructuredFieldSpecification */
+      field = collectionSpecSubset[fieldName],
+      fieldValues,
+      i,
+      fieldErrors,
+      result = true;
+
+    if (field.mandatory && !field.hasOwnProperty('defaultValue') && (!documentSubset.hasOwnProperty(fieldName) || typeof documentSubset[fieldName] === 'undefined' || documentSubset[fieldName] === null)) {
+      manageErrorMessage(field.path, errorMessages, 'The field is mandatory.', verbose);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 88 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)